src/test/objectstore/allocsim/ops_replayer.cc

```diff
@@ -33,6 +35,23 @@ namespace po = boost::program_options;
 using namespace std;
 using namespace ceph;
 
+namespace settings {
+
+// Returns a function which restricts a value to a specified range by throwing if it is not in range:
+// (Note: std::clamp() does not throw.)
+auto clamp_or_throw(auto min, int max)
+{
+ return [=](auto& x) { 
+		if(std::less<>{}(x, min) or std::greater<>{}(x, max)) {
```

Comment:
Another way to write this might be something like:
const auto y = std::clamp(x, min, max);
if(y != x) { throw; }
return y;

...I could go either way.